modules/bloop-rifle/src/main/scala/scala/build/blooprifle/BloopRifle.scala

```diff
@@ -47,7 +47,25 @@ object BloopRifle {
   ): Future[Unit] =
     config.classPath(version) match {
       case Left(ex) => Future.failed(new Exception("Error getting Bloop class path", ex))
-      case Right(cp) =>
+      case Right((cp, isScalaCliBloop)) =>
+        object IntValue {
+          def unapply(s: String): Option[Int] =
+            // no String.toIntOption in Scala 2.12.x
+            try Some(s.toInt)
+            catch {
+              case _: NumberFormatException => None
+            }
+        }
+        val bloopServerSupportsFileTruncating =
+          isScalaCliBloop && {
+            version.takeWhile(c => c.isDigit || c == '.').split('.') match {
+              case Array(IntValue(maj), IntValue(min), IntValue(patch)) =>
+                import scala.math.Ordering.Implicits._
+                Seq(maj, min, patch) >= Seq(1, 14, 20)
```

Comment:
I wonder why we do need to check the bloop version if we control the version of bloop ourselves? I think if we see too old version running (e.g. after update of scala-cli) we should restart it with newer version

Reply:
That's for if users manually force an earlier version to something else. Earlier Bloop versions can't truncate their output file, so redirecting the output this way would create a file that might grow too big.